controllers/chaosimpl/iochaos/impl.go

```diff
@@ -172,6 +175,12 @@ func (impl *Impl) Recover(ctx context.Context, index int, records []*v1alpha1.Re
 		if err == podiochaosmanager.ErrPodNotFound || err == podiochaosmanager.ErrPodNotRunning {
 			return v1alpha1.NotInjected, nil
 		}
+
+		if k8sError.IsForbidden(err) {
+			if strings.Contains(err.Error(), "because it is being terminated") {
+				return v1alpha1.NotInjected, nil
+			}
+		}
```

Comment:
That confuses me. It seems this `err` is returned by `Commit`, I saw that there are several actions like "Create", "Update", and "Get". But why does the termination could affect the RBAC authorization of PodIoChaos?

Reply:
The full error is: "Forbid to create xxx in namespace xxx, because it is being terminated". When the namespace is being deleted, `api-server` will refuse to create new object in that namespace, so that the `podiochaos` cannot be created, which is needed for recovery. However, it can be ignored safely, because when the original `podiochaos` is deleted, these pods have been recovered (if they still exist).

This error is really common in e2e test, and it will block e2e test to delete the working namespace.